runtime/ts/slot-composer.ts

```diff
@@ -51,29 +59,29 @@ export class SlotComposer {
     });
   }
 
-  get affordance() { return this._affordance.name; }
-  get consumers() { return this._consumers; }
+  get affordance(): string { return this._affordance.name; }
+  get consumers(): SlotConsumer[] { return this._consumers; }
 
-  getSlotConsumer(particle, slotName) {
-    return this.consumers.find(s => s.consumeConn.particle == particle && s.consumeConn.name == slotName);
+  getSlotConsumer(particle: Particle, slotName: string): SlotConsumer {
+    return this.consumers.find(s => s.consumeConn.particle === particle && s.consumeConn.name === slotName);
   }
 
-  findContainerByName(name) {
+  findContainerByName(name: string) {
     const contexts = this._contexts.filter(context => context.name === name);
-    if (contexts.length == 0) {
+    if (contexts.length === 0) {
+      // TODO this is a no-op, but throwing here breaks tests
```

Comment:
Flagging this

Reply:
switched to console.warn

Reply:
Should be addressed as an error condition at some point or at least documented in the return type at some point.